### PR TITLE
Turn Include Font mod stats into a LoadoutParameter, configure LO with results from analysis

### PR DESCRIPTION
**Pull request by robojumper**
The behavior where we include font mod stats was neat but always a bit controversial and people used very specific workarounds for this. This makes it a per-loadout setting so that people have control over how Loadout Optimizer calculates stats. The toggle is hidden unless the Loadout has Font mods.

This is a prerequisite for allowing the background analyzer to take user intent with these loadouts into account. If users couldn't control this behavior, Loadouts where tiers exceed T10 due to font mods would pretty much always show a "better stats available" note that users don't want to act on.

Adds some explanatory notes and tooltips to the relevant UI elements explaining the interaction of these features and ensures that the "Optimize Armor" button configures LO as desired.

### CODE REVIEW

**Comment by robojumper** [comment]
The main problem right now is that because this loads set tiers into LO in order to find upgrades

grafik

it'll save those much stricter stat constraints back into the Loadout when saving the better build. This is potentially not desired.

**Review by bhollis — APPROVED** [review]
Love it! I appreciate the tests as well. I had a little concern about having to modify the "input loadout" in `LoadoutItemCategorySection` but it's a good way to work it now, and I'll probably have to move things around a lot to display the current loadout anyway.

File: src/app/loadout-builder/loadout-builder-reducer.ts [hunk 1]
```diff
@@ -79,6 +82,12 @@ interface LoadoutBuilderConfiguration {
    */
   isEditingExistingLoadout: boolean;
 
+  /**
+   * If we are editing an existing loadout via the "better stats available"
+   * feature, this contains the stats we actually need to exceed.
+   */
+  strictUpgradesStatConstraints: ResolvedStatConstraint[] | undefined;
```
[the remark this answers is not in the document]
Comment (robojumper) [reply src/app/loadout-builder/loadout-builder-reducer.ts:86]: Yeah, that sounds reasonable. Right now the responsibility of changing the loadout to fit with the workflow is sort of split between the analyzer (which outputs new LoadoutParameters), the button (which performs the merging) and LO itself (which initializes some more things). The core tradeoff here is that if LO does more of this in initialization, then the analyzer needs to rely more on what happens internally in LO and buggy situations are more likely to happen where the analyzer finds better sets but LO comes up with slightly different parameters and fails to find any. Definitely warrants enough thought to refactor this in a follow-up.

**Comment by bhollis** [comment]
LGTM, feel free to merge when you're comfortable.

**Pull request merged**